test/test_custom_gates.jl

```diff
@@ -169,4 +169,119 @@ using BraketSimulator: DoubleExcitation, SingleExcitation, matrix_rep, Control
             end
         end
     end
+
+    @testset "Single excitation plus" begin
+        ϕ  = 3.56
+        nq = 2
+        instructions = [Instruction(H(), [0]), Instruction(H(), [1]), Instruction(SingleExcitationPlus(ϕ), [0, 1])]
+        # instructions for the gate decomposition (from PennyLane)
+        de_instructions = [Instruction(H(), [0]), Instruction(H(), [1]), Instruction(X(), [0]), Instruction(X(), [1]), Instruction(CPhaseShift(ϕ/2), [1, 0]), Instruction(X(), [0]), Instruction(X(), [1]), Instruction(CPhaseShift(ϕ/2), [0, 1]), Instruction(CNot(), [0, 1]), Instruction(Ry(ϕ/2), [0]), Instruction(CNot(), [1, 0]), Instruction(Ry(-ϕ/2), [0]), Instruction(CNot(), [1, 0]), Instruction(CNot(), [0, 1])]
+        # instructions for the matrix representation
+        u_instructions = [Instruction(H(), [0]), Instruction(H(), [1]), Instruction(Unitary(Matrix(matrix_rep(SingleExcitationPlus(ϕ)))), [1, 0])]
+        state_vector = 0.5 * [exp(im*ϕ/2),  cos(ϕ/2) - sin(ϕ/2), cos(ϕ/2) + sin(ϕ/2), exp(im*ϕ/2)]
+        probability_amplitudes = [0.2499999999999999, 0.35157642553610396, 0.14842357446389573, 0.2499999999999999]
```

Comment:
Where did these come from? PennyLane itself? Would be good to leave a comment if so